src/main/java/com/mraof/minestuck/entity/ai/AnimatedPanicGoal.java

```diff
@@ -1,29 +1,33 @@
 package com.mraof.minestuck.entity.ai;
 
-import com.mraof.minestuck.entity.AnimatedCreatureEntity;
+import com.mraof.minestuck.entity.AnimatedPathfinderMob;
+import com.mraof.minestuck.entity.animation.MobAnimation;
 import net.minecraft.world.entity.ai.goal.PanicGoal;
 
 public class AnimatedPanicGoal extends PanicGoal
 {
-	private final AnimatedCreatureEntity entity;
+	private final AnimatedPathfinderMob entity;
+	private final MobAnimation panicAnimation;
 	
-	public AnimatedPanicGoal(AnimatedCreatureEntity entity, double speedModifier)
+	public AnimatedPanicGoal(AnimatedPathfinderMob entity, double speedModifier, MobAnimation panicAnimation)
 	{
 		super(entity, speedModifier);
 		this.entity = entity;
+		this.panicAnimation = panicAnimation;
 	}
 	
 	@Override
 	public void start()
 	{
-		this.entity.setCurrentAction(AnimatedCreatureEntity.Actions.PANIC);
+		if(panicAnimation != null)
```

Comment:
Perhaps mark `panicAnimation` as nullable if it's intended to potentially be null. Or consider whether it should be allowed to be nullable at all.